adapters/redis/lib/opentelemetry/adapters/redis/patches/client.rb

```diff
@@ -0,0 +1,78 @@
+# frozen_string_literal: true
+
+# Copyright 2020 OpenTelemetry Authors
+#
+# SPDX-License-Identifier: Apache-2.0
+
+module OpenTelemetry
+  module Adapters
+    module Redis
+      module Patches
+        # Module to prepend to Redis::Client for instrumentation
+        module Client
+          def call(*args, &block)
+            return super(*args, &block) if disable_span_reporting?(*args)
+
+            response = nil
+
+            tracer.in_span(
+              Utils.format_command(args),
+              attributes: client_attributes.merge(
+                'db.statement' => Utils.format_statement(args)
+              ),
+              kind: :client
+            ) do |_span|
+              response = super(*args, &block)
+            end
+
+            response
+          end
+
+          def call_pipeline(*args, &block)
+            return super(*args, &block) if disable_span_reporting?(*args)
+
+            response = nil
+
+            tracer.in_span(
+              'pipeline',
+              attributes: client_attributes.merge(
+                'db.statement' => Utils.format_pipeline_statement(args)
+              ),
+              kind: :client
+            ) do |_span|
+              response = super(*args, &block)
+            end
+
+            response
+          end
+
+          # Override implementation (subclass) to determine per-connection
+          # span reporting rules.
+          def disable_span_reporting?(*_args)
+            false
+          end
```

Comment:
I don't see how we have per-connection span reporting. Either this module is prepended to the redis client, or the no-op version is. We either get all spans, or no spans. We already have the ability to do this by installing the instrumentation (with c.use), or not installing the instrumentation. Correct me if I'm mistaken on this point.

Reply:
@mwear "per-connection" may be misleading, but the arguments provided to this method would allow someone to create their own patch that would could disable spans for certain methods, keys, values, etc. That may not be a use case that needs to be supported and could be simplified/removed, but I kept it for parity with the functionality of the HTTP client adapters so far.

Reply:
I would err on the side of YAGNI at this stage in the project unless compelling use cases become apparent. My preference would be to remove it for now. That would also apply to the http client instrumentation

Reply:
@mwear Should the ability to provide a custom patch/middleware for adapters still exist? It was only used in tests/examples for overriding this method, but I'm not sure if there is another use case or if it is desired for dependency injection.